lim_test.py

```diff
@@ -0,0 +1,259 @@
+        else:
+            inv_error = abs(expected // (result - expected))
+        self.assertGreater(inv_error, 1/allowed_error)
+
+    def test_li_div(self):
+        '''test long_int_div'''
+        self.assertEqual(lim.long_int_div(self.float_big, self.long_big), 0)
+        self.assertEqual(lim.long_int_div(self.float_sm, self.float_big), 0)
+
+        result1 = lim.long_int_div(self.float_big, self.float_sm)
+        self.error_tester(result1, 10**400)
+
+        result2 = lim.long_int_div(self.long_big, -self.float_big)
+        self.error_tester(result2, -10**700)
+
+        result3 = lim.long_int_div(2, self.float_big)
+        self.error_tester(result3, 2e-300)
+
+        result4 = lim.long_int_div(-self.float_big, 5)
+        self.error_tester(result4, -2e+299)
+
+    def test_li_times(self):
+        '''test long_int_times'''
+        result1 = lim.long_int_times(self.long_big, self.float_big)
+        self.error_tester(result1, 10**1300)
+
+        result2 = lim.long_int_times(-4 * self.float_sm, self.float_sm)
+        self.error_tester(result2, -4e-200)
+    def test_li_pow(self):
+        '''test long_int_pow'''
+        result1 = lim.long_int_pow(self.long_big, 2.5)
+        self.error_tester(result1, 10**2500)
+
+
+class TestLITable(unittest.TestCase):
+    '''tests for all functions on a LongIntTable that aren't add and remove'''
+    def setUp(self):
+        self.table_a = lim.LongIntTable({3:3, 1:1, 2:2})
+        self.table_a1 = lim.LongIntTable({3:3, 1:1, 2:2})
+
+        self.table_b = lim.LongIntTable({1:1, 2:0, 3:3})
+        self.table_b1 = lim.LongIntTable({1:1, 2:0, 3:3})
+
+        self.table_c = lim.LongIntTable({1:1, 2:5, 3:3})
+
+        self.empty_table_a = lim.LongIntTable({})
+        self.empty_table_b = lim.LongIntTable({1:0, 2:0})
+
+    def tearDown(self):
+        del self.table_a
+        del self.table_a1
+
+        del self.table_b
+        del self.table_b1
+
+        del self.table_c
+
+        del self.empty_table_a
+        del self.empty_table_b
+
+    def test_values(self):
+        '''all the values_something functions do what they should'''
+        self.assertEqual(self.table_a.values(), [1, 2, 3])
+        self.assertEqual(self.table_b.values(), [1, 3])
+        self.assertEqual(self.empty_table_b.values(), [])
+
+        self.assertIsNone(self.empty_table_a.values_min())
+        self.assertIsNone(self.empty_table_a.values_max())
+        self.assertEqual(self.table_a.values_min(), 1)
+        self.assertEqual(self.table_a.values_max(), 3)
+
+        self.assertEqual(self.table_a.values_range(), (1, 3))
+        self.assertEqual(self.empty_table_b.values_range(), (None, None))
+
+    def test_frequency(self):
+        '''all the frequency_something functions do what they should'''
+        #confirms frequency function returns appropriate value or zero
+        self.assertEqual(self.table_a.frequency(2), (2, 2))
+        self.assertEqual(self.table_a.frequency(5), (5, 0))
```

Comment:
good additional test case, verifying what happens when the value does not exist. This is it's own test method.
test_frequency_when_out_of_range (or something like that)